### PR TITLE
refactor: define typescript type for AccountDelete transaction

### PR DESCRIPTION
**Pull request by mukulljangid**
## High Level Overview of Change
Defines typescript types for AccountDelete transaction.

### Context of Change
AccountDelete is an extension of CommonFields, and is exported in a type union with all other transaction types defined as Transaction.

We also will provide an optional verifyAccountDelete function for runtime form checking of AccountDelete transactions, should the users want to verify transaction form at runtime.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Test Plan
Adds tests for verifyAccountDelete, which provides runtime form verification for AccountDelete transactions

### CODE REVIEW

File: src/models/transactions/accountDelete.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    verifyBaseTransaction(tx)
+
+    if (tx.Destination === undefined)
+        throw new ValidationError("AccountDelete: missing field Destination")
+    
+    if (typeof tx.Destination !== 'string')
+        throw new ValidationError("AccountDelete: invalid Destination")
+
+    if (tx.DestinationTag !== undefined && typeof tx.DestinationTag !== 'number')
+        throw new ValidationError("AccountDelete: invalid DestinationTag")
```
Comment (mvadari) [inline src/models/transactions/accountDelete.ts:27]: Check the fee too - it should be 5 XRP

Comment (mukulljangid) [reply src/models/transactions/accountDelete.ts:27]: Done!

File: src/models/transactions/accountDelete.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { ValidationError } from "../../common/errors";
+import { BaseTransaction, verifyBaseTransaction } from "./common";
+
+export interface AccountDelete extends BaseTransaction {
+    TransactionType: "AccountDelete";
+    Destination: string;
```
Comment (natenichols) [inline src/models/transactions/accountDelete.ts:6]: I think we can add `Fee: 5` to this typescript definition. Fee should always be 5 in `AccountDelete` Transactions IIRC.

**Review by natenichols — APPROVED** [review]
Left one comment, nothing to hold up approval, though. LGTM 👍 

Just make sure you add that `Fee: 5`

**Review by intelliot — CHANGES_REQUESTED** [review]
The fee can change based on a vote of the validators. We should at least explore/consider the possibility of retrieving the current owner reserve from rippled and using that as the fee.

**Review by intelliot — APPROVED** [review]
Actually, I'm fine with removing the check from the model. We should consider a preflight check in the client, as well as ensuring that autofill does the right thing.

**Pull request merged**